src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php

```diff
@@ -620,4 +623,32 @@ private function getErrorCode(Exception $e): int
 
         return Response::HTTP_INTERNAL_SERVER_ERROR;
     }
+
+    /**
+     * This method will be removed in the next patch version. We rely on Cart ObjectModel to simplify the code
+     *
+     * @param int $cartId
+     * @param int $productId
+     * @param int|null $combinationId
+     *
+     * @return int
+     */
+    private function getProductGiftedQuantity(int $cartId, int $productId, ?int $combinationId): int
```

Comment:
Controllers are not supposed to use directly legacy code 😅  (and this function does seem to belong to controller layer, it's a function that could exist at ObjectModel level because it's useful)

Reply:
that's why I added a comment to explain that it's a code that will be removed in next patch. But ... I did a regular and clean command to get a product cartRules. I stashed it because I thought it's too much :D

Reply:
I'm fifty/fifty on this 😅 
need one more opinion